src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -130,6 +132,38 @@ def __init__(self, msConfig, logger=None):
         self.wfCounters = {'cleaned': {},
                            'archived': {'normalArchived': 0,
                                         'forceArchived': 0}}
+        self.globalLocks = set()
+
+    def getGlobalLocks(self):
+        """
+        Fetches the list of 'globalLocks' from wmstats server and the list of
+        'parentLocks' from request manager. Stores/updates the unified set in
+        the 'globalLocks' instance variable. Returns the resultant unified set.
+        :return: A union set of the 'globalLocks' and the 'parentLocks' lists
+        """
+        globalLocks = set()
+        parentLocks = set()
+        self.logger.info("Fetching globalLocks list from wmstats server.")
+        try:
+            uri = 'globallocks'
+            globalLocks.update(set(self.wmstatsSvc._getResult(uri, clearCache=True, verb="GET")))
```

Comment:
Todor, instead of a direct call to that private method. I would suggest to implement a public API on the WMStatsServer module and use it. If the server returns a flat list, you might even cast it to a set() to be returned.

One more comment on this line, I suppose you want to simply refresh the list of locks (not to update it based on the previous cycle), right?

Reply:
Hi @amaltaro I agree this is good to have a dedicated  API  for that, but I am trying to avoid touching major components of a central service with a midmonth patch. We must for sure fix this for the next release. The current method is working well so far.

Reply:
Experience tells me that something non-ideal that goes to production, remains in production!
So I'd rather have it right since the beginning.

Reply:
@amaltaro I totally agree with you here, and I am about to develop those two methods of course. In the meantime you may also start reviewing the rest of the proposed changes. I will write back here once I am ready with these two.

Reply:
Done